staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/virtualmachine.go

```diff
@@ -214,6 +214,29 @@ func (vm *VirtualMachine) IsActive(ctx context.Context) (bool, error) {
 	return false, nil
 }
 
+// Exists checks if VM exists and is not terminated
+func (vm *VirtualMachine) Exists(ctx context.Context) (bool, error) {
+	vmMoList, err := vm.Datacenter.GetVMMoList(ctx, []*VirtualMachine{vm}, []string{"summary.runtime.powerState"})
+	if err != nil {
+		klog.Errorf("Failed to get VM Managed object with property summary. err: +%v", err)
+		return false, err
+	}
+	// We check for VMs which are still available in vcenter and has not been terminated/removed from
+	// disk and hence we consider PoweredOn,PoweredOff and Suspended as alive states.
+	aliveStates := []types.VirtualMachinePowerState{
```

Comment:
We do not have any other power state other than 3 mentioned here. What is the objective of checking against them?

I think the main objective of this function `Exists` is to make sure VM is present on the inventory, regardless of the it's power state.

Reply:
Hi @divyenpatel , there are only 3 states for VM https://github.com/vmware/govmomi/blob/3decde69e9aa0bcc8a831e3f87acae26c2151c4a/vim25/types/enum.go#L4573-L4575
Also I cherry picked this changes from this commit https://github.com/kubernetes/kubernetes/pull/70291/files
Hemant explained here why he added this check https://github.com/kubernetes/kubernetes/pull/70291/files#r229045231
It's to protect against any other state in the future sdk